core/SDL.carp

```diff
@@ -205,6 +215,66 @@
             r (/= 0 (Int.bit-and state (SDL.Mouse.button SDL.Mouse.button-right)))]
         (SDL.MouseState.init x y l r))))
 
+  ;; Window Event
+  (defmodule WindowEvent
+    (defmodule WindowEventID
+      (register window-shown        SDL_WindowEventID "SDL_WINDOWEVENT_SHOWN")
+      (register window-exposed      SDL_WindowEventID "SDL_WINDOWEVENT_EXPOSED")
+      (register window-hidden       SDL_WindowEventID "SDL_WINDOWEVENT_HIDDEN")
+      (register window-moved        SDL_WindowEventID "SDL_WINDOWEVENT_MOVED")
+      (register window-resized      SDL_WindowEventID "SDL_WINDOWEVENT_RESIZED")
+      (register window-size-changed SDL_WindowEventID "SDL_WINDOWEVENT_SIZE_CHANGED")
+      (register window-minimized    SDL_WindowEventID "SDL_WINDOWEVENT_MINIMIZED")
+      (register window-mximized     SDL_WindowEventID "SDL_WINDOWEVENT_MAXIMIZED")
+      (register window-restored     SDL_WindowEventID "SDL_WINDOWEVENT_RESTORED")
+      (register window-enter        SDL_WindowEventID "SDL_WINDOWEVENT_ENTER")
+      (register window-leave        SDL_WindowEventID "SDL_WINDOWEVENT_LEAVE")
+      (register window-focus-gained SDL_WindowEventID "SDL_WINDOWEVENT_FOCUS_GAINED")
+      (register window-focus-lost   SDL_WindowEventID "SDL_WINDOWEVENT_FOCUS_LOST")
+      (register window-close        SDL_WindowEventID "SDL_WINDOWEVENT_CLOSE")
+      (register window-take-focus   SDL_WindowEventID "SDL_WINDOWEVENT_TAKE_FOCUS")
+      (register window-hit-test     SDL_WindowEventID "SDL_WINDOWEVENT_HIT_TEST")
+
+      (register = (Fn [SDL_WindowEventID SDL_WindowEventID] Bool))
```

Comment:
You don’t need to register it if you define it yourself, only if it’s defined elsewhere (i.e. in C)